nntrainer/utils/util_simd.cpp

```diff
@@ -40,6 +40,34 @@ void swish(const unsigned int N, float *X, float *Y, float *Z) {
 #endif
 }
 
+float max(const unsigned int N, float *X) {
+#ifdef USE_NEON
+  return nntrainer::neon::max(N, X);
+#else
+  std::vector<float> v(X, X + N);
+  return *std::max_element(v.begin(), v.end());
+#endif
+}
+
+void softmax(const unsigned int N, float *X, float *Y) {
+#ifdef USE_NEON
+  nntrainer::neon::softmax(N, X, Y);
+#else
+  unsigned int i = 0;
+  float sum = 0.f;
+  float max_x = max(N, X);
+  while (i < N) {
```

Comment:
Not in this PR, but you can also optimize further.
using omp and temporal buffer to save X[i] - max_x.